src/ember-resource.js

```diff
@@ -7,6 +7,14 @@
       getPath = Ember.Resource.getPath,
       set = Ember.set;
 
+  // Determine if we need to be back compatible with Ember < 1.12
+  var requiresEmberComputedPropertyFunction = false;
+  try {
+    Ember.computed({get: function() {}, set: function(key, value) {}});
+  } catch(ex) {
+    requiresEmberComputedPropertyFunction = true;
+  }
+
```

Comment:
What is the exception you are depending on here? If it's an assertion failure, would it not be silenced in production ember builds?